superset/sql_parse.py

```diff
@@ -378,3 +380,23 @@ def set_or_update_query_limit(self, new_limit: int, force: bool = False) -> str:
         for i in statement.tokens:
             str_res += str(i.value)
         return str_res
+
+
+def validate_filter_clause(clause: str) -> None:
+    if sqlparse.format(clause, strip_comments=True) != sqlparse.format(clause):
+        raise QueryClauseValidationException("Filter clause contains comment")
```

Comment:
@villebro is this intended ? this exception broke a number of our dashboards and seems to be a valid use case for the where clause sql. e.g. comment out different filter options while iterating on the chart